cypress/e2e/main.spec.js

```diff
@@ -0,0 +1,158 @@
+/// <reference types="Cypress" />
+
+context('Window', () => {
+    beforeEach(() => {
+        cy.visit('/page1/');
+    });
+
+    it('should add swup class to html element', () => {
+        cy.get('html').should('have.class', 'swup-enabled');
+        cy.shouldHaveH1('Page 1');
+    });
+
+    it('should prevent the default click event', () => {
+        let triggered = false;
+        let prevented = false;
+        cy.document().then(document => {
+            document.documentElement.addEventListener('click', (event) => {
+                triggered = true
+                prevented = event.defaultPrevented
+            });
+        });
+        cy.triggerClickOnLink('/page2/');
+        cy.wait(100);
+        cy.window().then(() => {
+            expect(triggered, 'event was not triggered').to.be.true;
+            expect(prevented, 'preventDefault() was not called').to.be.true;
+        });
+    });
+
+    it('should trigger a custom click event', () => {
+        let triggered = false;
+        cy.window().then(window => {
+            window.swup.on('clickLink', () => triggered = true);
+        });
+        cy.triggerClickOnLink('/page2/');
+        cy.wait(100);
+        cy.window().then(() => {
+            expect(triggered, 'event was not triggered').to.be.true;
+        });
+    });
+
+    it('should transition to other pages', () => {
+        cy.triggerClickOnLink('/page2/');
+        cy.wait(1000);
+        cy.shouldBeAtPage('/page2/');
+        cy.shouldHaveH1('Page 2');
+
+        cy.wait(1000);
+
+        cy.triggerClickOnLink('/page3/');
+        cy.wait(1000);
+        cy.shouldBeAtPage('/page3/');
+        cy.shouldHaveH1('Page 3');
+    });
+
+    it('should ignore links with data-no-swup attr', () => {
+        cy.shouldNativelyLoadPageAfterAction('/page4/', () => {
+            cy.get(`a[data-no-swup]`).first().click();
+        });
+    });
+
+    // it('should ignore clicks when meta key pressed', () => {
+    //     cy.triggerClickOnLink('/page2/', { metaKey: true });
+    //     cy.wait(500);
+    //     cy.shouldBeAtPage('/page1/');
+    //     cy.shouldHaveH1('Page 1');
+    // });
```

Comment:
Any reason for this to be commented out? Seems to pass fine for me. Is it causing trouble in CI?

Reply:
Exactly, it's passing fine locally but it was stalling CI. Not sure why but there might be a mismatch between Cypress browser versions, not sure. Maybe you can get it to work though...

Reply:
I think the fact that we're trying to open up a new window is the problem in CI. Maybe something like the below could work? Mocking the code used by swup to actually open the window, and just test that it gets called? 

```javascript
cy.window().then((window) => {
   let called = false;
   cy.stub(window, 'open', url => {
      called = true;
   }).as("popup");
});
```